### PR TITLE
Action completed dialog

### PR DESCRIPTION
**Pull request by kackogut**
# Description

Dialog displayed when action is completed by user. Data is now refreshed when user completes action, as before app was navigating user back and not refreshing the data

Fixes https://tree.taiga.io/project/jelgar-now-u-1/us/160

[action.webm](https://github.com/now-u/now-u-app/assets/23612157/be555133-65b7-46c1-9187-2ab812f0a6f2)

# Checklist:

## Creator

- [x] The target branch is main
- [ ] I have updated the unreleased section of the change log
- [x] I have reviewed the 'files changed' tab on github to ensure all changes are as expected
- [x] I have added someone to review the pr

## Reviewer

- [ ] I have verified the above are all completed
- [ ] I have run the code locally to ensure it fufills the requirements of the issue
- [ ] I have reviewed the 'files changed' and commented on any sections which I think are not needed, incorrect or could be improved

## After pull

- [ ] If appropriate I have closed the issue/ moved the trello card

### CODE REVIEW

File: lib/ui/dialogs/action/action_completed_dialog.dart [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+            onPressed: () => viewModel.onClose(),
+            icon: const Icon(
+              CustomIcons.ic_close,
+            ),
+          ),
+        ),
+        const SizedBox(height: 16.0),
+        Flexible(
+          child: Text(
+            'Thank you for taking action!',
```
Comment (kackogut) [inline lib/ui/dialogs/action/action_completed_dialog.dart:84]: Will update with translations, once the PR is merged

File: lib/ui/views/action_info/action_info_viewmodel.dart [hunk 2]
```diff
@@ -30,8 +31,8 @@ class ActionInfoViewModel extends FutureViewModel<Action> {
     try {
       await _causesService.completeAction(data!.id);
       setBusy(false);
-      // TODO Really?
-      _routerService.back();
+      _dialogService.showActionCompletedDialog();
+      notifySourceChanged();
```
Comment (kackogut) [inline lib/ui/views/action_info/action_info_viewmodel.dart:35]: @JElgar Could you please double check if this is correct way to refresh the data? This is working fine, but maybe there's some other method from Stacked that I should use?

File: lib/theme/assets.dart [hunk 3]
```diff
@@ -0,0 +1,3 @@
+abstract class Assets {
```
Comment (kackogut) [inline lib/theme/assets.dart:1]: @JElgar a helper class to not write `'assets/...png' inside view files, please let me know what do you think 🙌

File: lib/theme/spacings.dart [hunk 4]
```diff
@@ -0,0 +1,16 @@
+/**
+ * Spacing naming follow similar convention as Material Design colors,
+ * so that it's not necessary to have spacings like xxxxxsmall etc.
+ * Three main spacings are defined as `small`, `medium` and `large`
+ * and number next to it is the percentage value of the base size.
+ * For example `small_150` = `small_100` * 150%
+ */
+abstract class Spacing {
```
Comment (kackogut) [inline lib/theme/spacings.dart:8]: @JElgar Another helper class to unify spacings, please let me know if that's not an overkill

File: lib/generated/intl/messages_en.dart [hunk 5]
```diff
@@ -24,6 +24,12 @@ class MessageLookup extends MessageLookupByLibrary {
   static Map<String, Function> _notInlinedMessages(_) => <String, Function>{
```
Comment (JElgar) [inline lib/generated/intl/messages_en.dart:24]: Is it possible to exclude these generated files from the repo?

File: lib/theme/spacings.dart (hunk 4 above)
Comment (JElgar) [reply lib/theme/spacings.dart:8]: I think we already have something similar here: https://github.com/now-u/now-u-app/blob/main/lib/assets/constants.dart#L80. Potentially we should move that and the assets into that file?

File: lib/theme/assets.dart (hunk 3 above)
Comment (JElgar) [reply lib/theme/assets.dart:1]: Clean thank you!

File: lib/ui/views/action_info/action_info_viewmodel.dart (hunk 2 above)
Comment (JElgar) [reply lib/ui/views/action_info/action_info_viewmodel.dart:34]: I think the method we have been using is `notifyListeners` (which comes from stacked `BaseViewModel`)

e.g. https://github.com/now-u/now-u-app/blob/main/lib/ui/views/home/home_viewmodel.dart#L118

File: lib/theme/spacings.dart (hunk 4 above)
Comment (kackogut) [reply lib/theme/spacings.dart:8]: Ah I wasn't aware of that, thanks, I'll migrate it

File: lib/generated/intl/messages_en.dart (hunk 5 above)
Comment (kackogut) [reply lib/generated/intl/messages_en.dart:24]: Yep, I think so. I'll need to add generation to our Github action, I will add it as part of this the translations tickets. Already added it to the description

File: lib/ui/views/action_info/action_info_viewmodel.dart (hunk 2 above)
Comment (kackogut) [reply lib/ui/views/action_info/action_info_viewmodel.dart:34]: I've tried it as a first thing, but unfortunately that wasn't updating the screen

**Pull request merged**